### ISSUE TITLE
Input Validation

### ISSUE THREAD

**Issue by farmahini** [Kahbazi/MediatR.AspNetCore.Endpoints#1]
How can I add my input validations when I use this library?

**Comment by Kahbazi** [comment]
You can add an `IRequestPreProcessor` for validation which uses `FluentValidation` and have a middleware before `app.UseEndpoints(...)` to catch `ValidationException` and return the error response you want.
I will include this in the sample code.

The only thing I don't like about this is that MediatR throws an `AggregateException`.

```csharp
public class ValidationBehavior<TRequest> : IRequestPreProcessor<TRequest>
{
    private readonly IEnumerable<IValidator<TRequest>> _validators;

    public ValidationBehavior(IEnumerable<IValidator<TRequest>> validators)
    {
        _validators = validators;
    }

    public Task Process(TRequest request, CancellationToken cancellationToken)
    {
        var context = new ValidationContext(request);
        var failures = _validators
            .Select(v => v.Validate(context))
            .SelectMany(result => result.Errors)
            .Where(f => f != null)
            .ToList();

        if (failures.Count != 0)
        {
            throw new ValidationException(failures);
        }

        return Task.CompletedTask;
    }
}
```

```csharp
app.Use(async (context, next) =>
{
    try
    {
        await next();
    }
    catch (AggregateException aggregateException) when (aggregateException.InnerExceptions[0] is ValidationException validationException)
    {
        context.Response.StatusCode = 400;
    }
});
```

**Comment by Kahbazi** [comment]
> The only thing I don't like about this is that MediatR throws an AggregateException.

It looks like a bug, since the generic Send method does not throw an `AggregateException`.
I filed an [issue](https://github.com/jbogard/MediatR/issues/480) on MediatR repo and I will update the sample when this is fixed.

**Comment by Kahbazi** [comment]
This bug has been fixed in MediatR 8.0.1 . I've added a sample with input validation in samples folder.
